### PR TITLE
feat: UnsupervisedLOLDataLoader

### PR DESCRIPTION
**Pull request by soumik12345**
Fixes #<>

Changes proposed by this PR:

- [x] Add `UnsupervisedLOLDataLoader`
- [ ] ~Simplify example notebooks~

### CODE REVIEW

File: restorers/dataloader/base/base_dataloader.py [hunk 1]
```diff
@@ -168,7 +168,7 @@ def build_dataset(
         dataset = dataset.batch(batch_size, drop_remainder=True)
         return dataset.prefetch(_AUTOTUNE)
 
-    def get_datasets(self, batch_size: int) -> Tuple[tf.data.Dataset]:
+    def get_datasets(self, batch_size: int) -> Tuple[tf.data.Dataset, tf.data.Dataset]:
         """
         Function to retrieve a dataset.
```
Comment (ariG23498) [inline restorers/dataloader/base/base_dataloader.py:173]:
```suggestion
        Function to retrieve the train and val dataset.
```

File: restorers/dataloader/base/commons.py [hunk 2]
```diff
@@ -12,6 +12,14 @@ def random_horiontal_flip(low_image, enhanced_image):
     )
 
 
+def random_unpaired_horiontal_flip(low_image):
```
Comment (ariG23498) [inline restorers/dataloader/base/commons.py:15]: Just wanted to add that with the random flips ([left-right](https://www.tensorflow.org/api_docs/python/tf/image/random_flip_left_right)/[up-down](https://www.tensorflow.org/api_docs/python/tf/image/random_flip_up_down)) in the `tf.image` library, we can specify the same `seed` argument and both the images (low and enhanced) will be treated with the same augmentation.

We might want to benchmark the two APIs (the present pipeline with `tf.cond`, and the one with the `seed`).

**Review by ariG23498 — CHANGES_REQUESTED** [review]
Really liked the implementation and the tests.

File: restorers/dataloader/base/commons.py (hunk 2 above)
Comment (soumik12345) [reply restorers/dataloader/base/commons.py:15]: @ariG23498 Let's convert this comment into an issue.

Comment (soumik12345) [reply restorers/dataloader/base/commons.py:15]: https://github.com/TrystAI/restorers/issues/59

**Comment by soumik12345** [comment]
Fixes https://github.com/TrystAI/restorers/issues/59

**Review by SauravMaheshkar — APPROVED** [review]

**Review by ariG23498 — APPROVED** [review]
LGTM!
